src/spectral.ts

```diff
@@ -52,7 +53,16 @@ export class Spectral {
 
   constructor(protected readonly opts?: IConstructorOpts) {
     this._computeFingerprint = memoize(opts?.computeFingerprint ?? defaultComputeResultFingerprint);
-    this._resolver = opts?.resolver ?? new Resolver();
+    if (opts?.proxyUri) {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const ProxyAgent = require('proxy-agent');
+      DEFAULT_REQUEST_OPTIONS.agent = new ProxyAgent(opts.proxyUri);
```

Comment:
The problem with this approach is that you cannot have more than a single instance of Spectral, one that would be proxied and the other that wouldn't, since this setting is global.
There is a small trick we could do, however.
How about having `createHttpAndFileResolver` fn in `src/resolvers/http-and-file.ts` that would take custom request options?
We could pass the whole agent there and use the resolver returned by that fn.
This way, proxy agent would not be shared among all Spectral instances.

Reply:
Note - it's totally fine if you don't have spare time for doing this.
I'm happy to pick this up and get it over the finish line.
You've already done the vast chunk of the whole task anyway - thanks again!
If you are planning to work on this, no rush, take your time.

Reply:
Oh good point, I didn't think of that. No worries, I can fix that up. Thanks.

Reply:
Thank you!
Shall you have any questions, do feel free to reach out.

Reply:
Updated. Let me know if there's anything else that needs fixing or can be improved. I should have a bit of time this week so happy to fix! :beers: